web/src/actions/state.js

```diff
@@ -0,0 +1,71 @@
+import axios from '../util/api';
+
+export const GET_STATE_REQUEST = 'GET_STATE_REQUEST';
+export const GET_STATE_SUCCESS = 'GET_STATE_SUCCESS';
+export const GET_STATE_FAILURE = 'GET_STATE_FAILURE';
+
+export const UPDATE_STATE_REQUEST = 'UPDATE_STATE_REQUEST';
+export const UPDATE_STATE_SUCCESS = 'UPDATE_STATE_SUCCESS';
+export const UPDATE_STATE_FAILURE = 'UPDATE_STATE_FAILURE';
+
+export const requestState = () => ({ type: GET_STATE_REQUEST });
+export const receiveState = data => ({ type: GET_STATE_SUCCESS, data });
+export const failState = error => ({ type: GET_STATE_FAILURE, error });
+
+export const requestStateUpdate = () => ({ type: UPDATE_STATE_REQUEST });
+export const receiveStateUpdate = data => ({
+  type: UPDATE_STATE_SUCCESS,
+  data
+});
+export const failStateUpdate = error => ({ type: UPDATE_STATE_FAILURE, error });
+
+export const fetchState = id => dispatch => {
+  dispatch(requestState());
+
+  const url = `/states${id ? `/${id}` : ''}`;
+
+  return axios
+    .get(url)
+    .then(req => dispatch(receiveState(req.data)))
+    .catch(error => {
+      const reason = error.response.data || 'N/A';
```

Comment:
we can't rely on `error.response` being there; because of that, i did `const reason = error.response ? error.response.data : 'N/A';` in `auth.js`, though now that this has come up again, i wonder if we should add a utility method (`handleApiError`?) and use that in all these catches (also because i'm sure we're gonna want to change the `"N/A"` bit at some point